File: frontend/src/types/reports.ts
export type ReportSourcePerformance = {
  source: string;
  leads: number;
  won_opportunities: number;
  lost_opportunities: number;
  conversion_rate: number;
  won_value: string;
  lead_share_percent: number;
};

export type ReportMemberOption = {
  public_id: string;
  name: string;
  role: string;
};

export type ReportMemberPerformance = ReportMemberOption & {
  leads: number;
  open_opportunities: number;
  won_opportunities: number;
  lost_opportunities: number;
  conversion_rate: number;
  won_value: string;
  average_ticket: string;
  completed_activities: number;
};

export type ReportPipelineStage = {
  stage: string;
  opportunities: number;
  value: string;
  share_percent: number;
};

export type ReportLossReason = {
  reason: string;
  count: number;
  percentage: number;
};

export type ReportInterestPerformance = {
  interest: string;
  leads: number;
  won_opportunities: number;
  won_value: string;
};

export type ReportRevenuePoint = {
  period_start: string;
  period_end: string;
  amount: string;
  won_opportunities: number;
};

export type ReportAnalytics = {
  workspace_public_id: string;
  workspace_name: string;
  generated_at: string;
  period_days: number;
  period_start: string;
  period_end: string;
  source_filter: string | null;
  owner_user_public_id_filter: string | null;

  total_leads: number;
  opportunities_created: number;
  open_opportunities: number;
  open_pipeline_value: string;
  won_opportunities: number;
  lost_opportunities: number;
  conversion_rate: number;
  won_value: string;
  average_ticket: string;

  activities_created: number;
  completed_activities: number;
  pending_activities: number;
  overdue_activities: number;

  available_sources: string[];
  available_members: ReportMemberOption[];
  source_performance: ReportSourcePerformance[];
  member_performance: ReportMemberPerformance[];
  pipeline_stages: ReportPipelineStage[];
  loss_reasons: ReportLossReason[];
  interest_performance: ReportInterestPerformance[];
  revenue_series: ReportRevenuePoint[];
};

File: app/schemas/reports.py
from __future__ import annotations

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class ReportSourcePerformanceRead(BaseModel):
    source: str
    leads: int
    won_opportunities: int
    lost_opportunities: int
    conversion_rate: float
    won_value: Decimal
    lead_share_percent: float


class ReportMemberOptionRead(BaseModel):
    public_id: str
    name: str
    role: str


class ReportMemberPerformanceRead(BaseModel):
    public_id: str
    name: str
    role: str
    leads: int
    open_opportunities: int
    won_opportunities: int
    lost_opportunities: int
    conversion_rate: float
    won_value: Decimal
    average_ticket: Decimal
    completed_activities: int


class ReportPipelineStageRead(BaseModel):
    stage: str
    opportunities: int
    value: Decimal
    share_percent: float


class ReportLossReasonRead(BaseModel):
    reason: str
    count: int
    percentage: float


class ReportInterestPerformanceRead(BaseModel):
    interest: str
    leads: int
    won_opportunities: int
    won_value: Decimal


class ReportRevenuePointRead(BaseModel):
    period_start: datetime
    period_end: datetime
    amount: Decimal
    won_opportunities: int


class ReportAnalyticsRead(BaseModel):
    workspace_public_id: str
    workspace_name: str
    generated_at: datetime
    period_days: int
    period_start: datetime
    period_end: datetime
    source_filter: str | None
    owner_user_public_id_filter: str | None

    total_leads: int
    opportunities_created: int
    open_opportunities: int
    open_pipeline_value: Decimal
    won_opportunities: int
    lost_opportunities: int
    conversion_rate: float
    won_value: Decimal
    average_ticket: Decimal

    activities_created: int
    completed_activities: int
    pending_activities: int
    overdue_activities: int

    available_sources: list[str]
    available_members: list[ReportMemberOptionRead]
    source_performance: list[ReportSourcePerformanceRead]
    member_performance: list[ReportMemberPerformanceRead]
    pipeline_stages: list[ReportPipelineStageRead]
    loss_reasons: list[ReportLossReasonRead]
    interest_performance: list[ReportInterestPerformanceRead]
    revenue_series: list[ReportRevenuePointRead]
